### DISCUSSION TITLE
updating kernel after the booting with dpkg

### DISCUSSION THREAD

**Discussion by Ruach** [Joshua-Riek/ubuntu-rockchip#847] [Q&A]
Hi I could successfully flash the image with custom built linux (#844).
As it is quite time consuming to generate new image, which also get rid of entire content of the SD-CARD, I would like to understand if it is possible to just update the kernel image through dpkg after booting up the rk3588 like below command.

$ dpkg -i linux-image-6.1.43-rockchip_6.1.43-rockchip-1_arm64.deb

The dpkg was successfully installed, but the orangepi doesn't boot up after that. Do I need get rid of previously installed image before dpkg new kernel image? Or what additional operations are necessary? 

Thanks!

**Comment by Joshua-Riek** [comment]
You did not compile the kernel correctly.. I use upstream Ubuntu packaging..
See: https://wiki.ubuntu.com/Kernel/BuildYourOwnKernel

Or look at the source code to compile the kernel..
```
export $(dpkg-architecture -aarm64)
export CROSS_COMPILE=aarch64-linux-gnu-
export CC=aarch64-linux-gnu-gcc
export LANG=C

# Compile the kernel into a deb package
fakeroot debian/rules clean binary-headers binary-rockchip do_mainline_build=true
```

**Reply by PhilZProg** [reply]
Hi!
Im trying to change some kernel options and I followed instructions here. Im not able to do any customizations as `fakeroot debian/rules editconfigs` are not present and if i do `make olddefconfig` with `make menuconfig` afterwords the build exiting with error. What is the correct way to customize and install kernel?

**Reply by Joshua-Riek** [reply]
modify the config in debian.rockchip/config/config.common.ubuntu

**Reply by Ruach** [reply]
@Joshua-Riek  HI Joshua, what do you mean by I didn't compile the kernel correctly? is it because that I mentioned only on deb package? Actually I have several ones and got installed all of them after booting orangepi

**Reply by PhilZProg** [reply]
Thank you for the tip about config.common.ubuntu. Is there any way I cane modify this config though menuconfig? Or do I have to work with config.common.ubuntu through some text editor only?
